diff --git a/src/libtools/signal_private.h b/src/libtools/signal_private.h
index 8923119..e11d95e 100644
--- a/src/libtools/signal_private.h
+++ b/src/libtools/signal_private.h
@@ -185,7 +185,7 @@ typedef struct x64_mcontext_s
 } x64_mcontext_t;

 // /!\ signal sig_set is different than glibc __sig_set
-#ifndef ANDROID
+#ifndef _NSIG_WORDS
 #define _NSIG_WORDS (1024 / (sizeof(unsigned long int)*8))
 #endif
